docs/custom-tests/05-bootstrap-and-test-phases.md

```diff
@@ -0,0 +1,132 @@
+
+**bootstrap.sh**
+
+```bash
+# Install Storefront Theme so that we can activate it on our Test phase.
+wp theme install storefront
+```
+
+### `bootstrap.sh`
+
+This is a bash script that, if present, will be called. Here you can use `wp` (WP-CLI) and bash in general.
+
+### `bootstrap.php`
+
+This is a PHP file that, if present, will be called. WordPress is NOT loaded on the context of this file.
+
+### `must-use-plugin.php`
+
+This is a file that, if present, will be copied to the `wp-content/mu-plugins` directory and will run on ALL requests on the E2E test. WordPress is loaded on the context of this file.
+
+If you want to put logic in there that should run only once, make sure to set a flag to prevent it from running multiple times.
+
+### Where to Put the Bootstrap Files
+
+Place them in the bootstrap directory within your E2E test suite. Here’s how your directory structure should look:
+
+```
+bootstrap             (Optional)
+  bootstrap.php       (Optional)
+  bootstrap.sh        (Optional)
+  must-use-plugin.php (Optional)
+entrypoint.qit.js     (Optional)
+example.spec.js
+```
+
+**You can put the tests anywhere you want**, even outside the plugin structure. Just for example purposes, here's what a typical directory structure could look like, using the `qit-beaver` plugin as an example:
+
+```
+.
+├── tests                            # Test files directory
+│    └── e2e                         # End-to-end test suite
+│        ├── bootstrap               # Bootstrap files for setting up tests
+│        │   ├── bootstrap.php       
+│        │   ├── bootstrap.sh        
+│        │   └── must-use-plugin.php
+│        ├── entrypoint.qit.js       # The Test Phase entrypoint 
+│        └── example.spec.js         # E2E test file
+├── src                              # QIT Beaver Source code
+└── qit-beaver.php                   # Main plugin file
+```
+
+## The Entrypoint
+
+The `entrypoint.spec.js` is a file that, if present, will be called when your **Test Phase** starts. This is a good place to put any setup logic, such as **activating** a theme that you have previously **installed** in the Bootstrap Phase.
+
+## The Test Phase
+
+This is where the actual testing happens. We utilize Playwright to interact with the browser and verify your plugin's functionality. By default, the only test phase that gets executes is the one from the plugin you are running the tests from.
+
+:::tip
+When running with the test phase of other plugins, any flakiness in the tests of other plugins can cause your tests to fail.
+:::
+
+### E2E Test
+
+If you run this command: `qit run:e2e qit-the-beaver --plugin cat-pictures`:
+
+- Run the bootstrap phases of all the plugins
+- **Run the test phase of `qit-the-beaver`**
+
+### Compatibility Test
+
+If instead you run with compatibility mode full, `qiit run:e2e qit-the-beaver --plugins cat-pictures --compatibility_mode=full`, it will:
```

Comment:
```suggestion
If instead you run with compatibility mode full, `qiit run:e2e qit-the-beaver --plugin cat-pictures --compatibility_mode=full`, it will:
```